### PR TITLE
Support nested html page analysis #43

### PR DESCRIPTION
**Pull request by prix-uniris**
It Supports Nested Html Files and Folders. 

Since file folder dirs can be pretty big, using to calculate a sum of transaction costs and asking user consent for that instead of asking permission for each file.

### CODE REVIEW

**Review by manujuniris — COMMENTED** [review]
Can you display number of replications ?

File: commands/deploy_website.js [hunk 1]
```diff
@@ -1,226 +1,260 @@
...
+              full_file_path: abs,
+            };
+            return;
+          } else {
+            path_struct[File] = {
+              seed: "",
+              address: "",
+              full_file_path: abs,
+            };
+            return;
```
Comment (blackode) [inline commands/deploy_website.js:67]: You can abstract this to a function and re-use it as instead repeating
```suggestion
            function pathStructure(filePath) {
              {
                seed: "",
                address: "",
                full_file_path: filePath
              }
             }
```
and just call something like `path_struct[key] = pathStructure(abs)` 

what do you think? :)

**Review by blackode — CHANGES_REQUESTED** [review]

**Comment by prix-uniris** [comment]
Made the changes above mentioned.

**Comment by manujuniris** [comment]
> Made the changes above mentioned.

Screenshot from 2022-04-04 13-15-54

I guess this is too much information to display for cli, the only things we should display must be following -
1. Transaction fee for entire folder
2. File path
3. Transaction URL 
4. Confirmation 

Also make use of chalk https://www.npmjs.com/package/chalk

**Comment by prix-uniris** [comment]
I have worked on the feedback.

File: commands/deploy_website.js (hunk 1 above)
Comment (prix-uniris) [reply commands/deploy_website.js]: Sure thanks, Applied it.

Comment (prix-uniris) [reply commands/deploy_website.js]: Sure thanks, Applied it.

**Review by manujuniris — APPROVED** [review]
Following output was captured while testing this PR - 

Screenshot from 2022-04-05 09-59-51

**Comment by prix-uniris** [comment]
Review Points for this Issue:
- Detection of nested files from the folder (.css, .js, .pngs, etc.)
- Upload those files to the blockchain
- Replace all the links in all HTML pages (index.html, about.html) which point to local links -> URLs of content on the blockchain.
- Website properly working on blockchain. With proper links to other pages, and images/assets.

**Review by blackode — APPROVED** [review]
- [x] Read PR description
- [x] Read AC
- [x] Pulled branch, manually tested, verified AC are met
- [x] Reviewed the code
- [x] Detection of nested files from the folder (.css, .js, .pngs, etc.)
- [x] Aggregate Transactin fee
image

- [x] Upload those files to the blockchain
- [x] Replace all the links in all HTML pages (index.html, about.html) which point to local links -> URLs of content on the blockchain.
- [x] Website properly working on blockchain. With proper links to other pages, and images/assets
image
-[x] Displaying missing index.html
image

.

**Comment by blackode** [comment]
LGTM

**Pull request merged**